### ISSUE TITLE
`dotnet.exe` doesn't uniquely identify duplicate resolved SDK paths

### ISSUE THREAD

**Issue by deeprobin** [dotnet/runtime#69371]
I noticed that the `dotnet.exe` detects the same SDK several times, because there are 2 different versions to the dotnet path in my `PATH` variable.

![grafik](https://user-images.githubusercontent.com/10373452/168402624-ef40f0eb-026c-4fd4-96ab-3d77a6f062b5.png)

### To Reproduce
Add to the `PATH` env the following values:
* `C:\Program Files\dotnet\`
* `C:\PROGRA~1\dotnet\`

Of course, one can expect that a user usually does not play around with the environment variables senselessly. However, in my case I had to reduce the length of my path and with each installation the written out version is added.

### Further technical details
```
.NET SDK (gemäß "global.json"):
 Version:   7.0.100-preview.4.22252.9
 Commit:    6bc31786a8
 ```

**Comment by AaronRobinsonMSFT** [comment]
Eww. This is relying on the Windows 8.3 short path logic. I'm not convinced there is a bug here to be honest. The same behavior would likely be produced using file system links.

A potential option to mitigate this would be to fully resolve each SDK path to determine if they resolve to the same directory. This would be rather difficult because NTFS has so many mechanisms that would need to be handled.

**Comment by agocke** [comment]
Yeah... I don't think this is worth fixing. Given the long bug tail, regression risk, and limited benefit (I think this is the first time anyone's ever reported this, but the bug has probably been around for many releases), I can't see scheduling this work ahead of many other things we'd like to improve.

I'm going to close this as wontfix.

**Comment by vitek-karas** [comment]
I didn't try but I think this will only happen if the `dotnet.exe` is resolved from the 8.3 path. If you swap the paths in your `PATH` so that `dotnet.exe` resolved from the full path I don't think this will repro.
